src/widgets/heightandweight/heightandweight-detailed-summary.component.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from "react";
+import { match, useParams } from "react-router";
+import SummaryCard from "../cards/summary-card.component";
+import styles from "./heightandweight-detailed-summary.css";
+import dayjs from "dayjs";
+import { getDimenionsObservationsRestAPI } from "./heightandweight.resource";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import { isEmpty } from "lodash-es";
+import {
+  convertToPounds,
+  convertToFeet,
+  convertoToInches
+} from "./heightandweight-helper";
+
+export function HeightAndWeightDetailedSummary(
+  props: HeightAndWeightDetailedSummaryProps
+) {
+  const [dimensions, setDimensions] = useState<any>({});
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+  let params = useParams();
+
+  useEffect(() => {
+    getDimenionsObservationsRestAPI(patientUuid).subscribe(response => {
+      setDimensions(
+        response.find(
+          dimension => dimension.obsData.weight.uuid === params["uuid"]
+        )
+      );
+    });
+  }, [params, patientUuid]);
+
+  function displayNoHeightAndWeight() {
+    return (
+      <SummaryCard
+        name="Height & Weight"
+        match={props.match}
+        styles={{ width: "100%" }}
+      >
+        <div className={styles.heightAndWeightDetailedSummary}>
+          <p className="omrs-bold" data-testid="errorMessage">
```

Comment:
see comment on #117 about using `data-testid`

Reply:
fixed the test